chain/chain/src/chain.rs

```diff
@@ -2358,6 +2370,12 @@ impl Chain {
         // Check if block can be finalized and drop it otherwise.
         self.check_if_finalizable(header)?;
 
```

Comment:
please add check that rejects blocks with `SpiceBlockBodyV3`  if `protocol_feature_spice` is not enabled, so we won't have maliciously created spice blocks in mainnet

Reply:
Great catch! Added in [9da4f20](https://github.com/near/nearcore/pull/13970/commits/9da4f20c6bed8fb987787f1642a4a153fa8e1f58)